projects/packages/jetpack-mu-wpcom/src/features/wpcom-admin-menu/wpcom-admin-menu.php

```diff
@@ -148,31 +147,142 @@ function wpcom_add_hosting_menu() {
  * Adds WordPress.com submenu items related to Jetpack under the Jetpack admin menu.
  */
 function wpcom_add_jetpack_submenu() {
-	/*
-	 * Do not display any menu on WoA and WordPress.com Simple sites (unless Classic wp-admin is enabled).
-	 * They already get a menu item under Users via nav-unification.
-	 */
-	if ( ( new Host() )->is_wpcom_platform() && get_option( 'wpcom_admin_interface' ) !== 'wp-admin' ) {
+	$is_simple_site          = defined( 'IS_WPCOM' ) && IS_WPCOM;
+	$is_atomic_site          = ! $is_simple_site;
+	$uses_wp_admin_interface = get_option( 'wpcom_admin_interface' ) === 'wp-admin';
+	$domain                  = wp_parse_url( home_url(), PHP_URL_HOST );
+
+	if ( ! $uses_wp_admin_interface ) {
+		return;
+	}
+
+	if ( $is_atomic_site && ( ( new Status() )->is_offline_mode() || ! ( new Connection_Manager( 'jetpack' ) )->is_user_connected() ) ) {
 		return;
 	}
 
-	/*
-	 * Do not display if we're in Offline mode, or if the user is not connected.
-	 */
-	if ( ( new Status() )->is_offline_mode() || ! ( new Connection_Manager( 'jetpack' ) )->is_user_connected() ) {
+	$blog_id = Connection_Manager::get_site_id( true );
+	if ( is_wp_error( $blog_id ) ) {
 		return;
 	}
 
+	// Hide submenu items that link to Jetpack Cloud.
+	wpcom_hide_submenu_page( 'jetpack', esc_url( Redirect::get_url( 'cloud-activity-log-wp-menu', array( 'site' => $blog_id ) ) ) );
+	wpcom_hide_submenu_page( 'jetpack', esc_url( Redirect::get_url( 'cloud-scan-history-wp-menu' ) ) );
+	wpcom_hide_submenu_page( 'jetpack', esc_url( Redirect::get_url( 'calypso-backups' ) ) );
+	wpcom_hide_submenu_page( 'jetpack', esc_url( Redirect::get_url( 'jetpack-menu-jetpack-manage-subscribers', array( 'site' => $blog_id ) ) ) );
+
+	$activity_log_url = 'https://wordpress.com/activity-log/' . $domain;
```

Comment:
[Minor] We could move initializing the domain here so that if some checks don't pass we don't initialize a variable that won't be used.